R/recipe.R

```diff
@@ -16,11 +16,8 @@ vetiver_create_meta.recipe <- function(model, metadata) {
 #' @rdname vetiver_create_ptype
 #' @export
 vetiver_ptype.recipe <- function(model, ...) {
-    rlang::check_dots_used()
-    dots <- list(...)
-    check_ptype_data(dots)
-    ptype <- vctrs::vec_ptype(dots$prototype_data)
-    tibble::as_tibble(ptype)
+    preds <- vec_slice(model$var_info, model$var_info$role == "predictor")
+    model$ptype[preds$variable]
```

Comment:
This is not in line with what recipes expects at bake time. As per [?update_role_requirements()](https://recipes.tidymodels.org/reference/update_role_requirements.html)

The following represents the default bake time requirements of specific types of roles:

- `"outcome"`: Not required at bake time. Can't be changed.

- `"predictor"`: Required at bake time. Can't be changed.

- `"case_weights"`: Not required at bake time by default.

- `NA`: Required at bake time by default.

- Custom roles: Required at bake time by default.

The above code fails for `NA` roles and custom roles.

``` r
data_orig <- tibble::tibble(
  y = 1:10,
  id = 1:10,
  x1 = letters[1:10],
  x2 = factor(letters[1:10]),
  cw = hardhat::importance_weights(1:10)
)

library(recipes)

model <- recipe(data_orig) |>
  update_role(y, new_role = "outcome") |>
  update_role(x1, x2, new_role = "predictor") |>
  update_role(id, new_role = "keep_at_bake") |>
  prep()

library(vctrs)

preds <- vec_slice(model$var_info, model$var_info$role == "predictor")

model$ptype[preds$variable]
#> # A tibble: 0 × 2
#> # ℹ 2 variables: x1 <chr>, x2 <fct>

recipes::recipes_ptype(model, stage = "bake")
#> # A tibble: 0 × 3
#> # ℹ 3 variables: id <int>, x1 <chr>, x2 <fct>
```

I would recommend that you use `recipes::recipes_ptype()` instead
```suggestion
    recipes::recipes_ptype(model, stage = "bake")
```

Reply:
Right right right, I am remembering now. Thanks!

Reply:
i only remember because we have it written down 😅